api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/integration/SequentialInferenceTest.kt

```diff
@@ -370,10 +374,7 @@ class SequentialInferenceTest {
         val model = Sequential.loadDefaultModelConfiguration(tempDir.toFile())
 
         model.use {
-            for (layer in it.layers) {
-                if (layer::class == Conv2D::class)
-                    layer.isTrainable = false
-            }
+            it.layers.filterIsInstance<Conv2D>().forEach(Layer::freeze)
```

Comment:
filterIsInstanse with generic looks worse than we removed a piece of code

How is it implemented internally? Does it iterate two times over the list of layers?

Reply:
Yes, it does iterate two times for each `Conv2D` layer. Imao, the model probably does not have that many convolutional layers when it could become a problem.